samples/dnn/text_detection.cpp

```diff
@@ -33,26 +38,35 @@ int main(int argc, char** argv)
     float nmsThreshold = parser.get<float>("nms");
     int inpWidth = parser.get<int>("width");
     int inpHeight = parser.get<int>("height");
-    String model = parser.get<String>("model");
+    String model_decoder = parser.get<String>("model");
+    String model_recognition = parser.get<String>("ocr");
 
     if (!parser.check())
     {
         parser.printErrors();
         return 1;
     }
 
-    CV_Assert(!model.empty());
+    CV_Assert(!model_decoder.empty());
+
+    // Load networks.
+    Net detector = readNet(model_decoder);
+    Net recognizer;
 
-    // Load network.
-    Net net = readNet(model);
+    if (!model_recognition.empty())
+        recognizer = readNet(model_recognition);
 
     // Open a video file or an image file or a camera stream.
     VideoCapture cap;
     if (parser.has("input"))
-        cap.open(parser.get<String>("input"));
+    {
+        bool open_success = cap.open(parser.get<String>("input"));
+        CV_Assert(open_success);
```

Comment:
Another branch with `cap.open(0);` needs similar check too. So it is better to move it down after this `if`.